experimental/web_dashboard/lib/src/api/mock.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:uuid/uuid.dart' as uuid;
+
+import 'api.dart';
+
+class MockDashboardApi implements DashboardApi {
+  @override
+  final EntryApi entries = MockEntryApi();
+
+  @override
+  final ItemApi items = MockItemApi();
+}
+
+class MockItemApi implements ItemApi {
+  Map<String, Item> _storage = {};
+  StreamController<List<Item>> _streamController =
+      StreamController<List<Item>>.broadcast();
+
+  @override
+  Future<Item> delete(String id) async {
+    _broadcast();
+    return _storage.remove(id);
+  }
+
+  @override
+  Future<Item> get(String id) async {
+    return _storage[id];
+  }
+
+  @override
+  Future<Item> insert(Item item) async {
+    var id = uuid.Uuid().v4();
+    var newItem = Item(item.name)..id = id;
+    _storage[id] = newItem;
+    _broadcast();
+    return newItem;
+  }
+
+  @override
+  Future<List<Item>> list() async {
+    return _storage.values.toList();
+  }
+
+  @override
+  Future<Item> update(Item item, String id) async {
+    _storage[id] = item;
+    return item..id = id;
+  }
+
+  Stream<List<Item>> subscribe() {
+    return _streamController.stream;
+  }
+
+  void _broadcast() {
+    _streamController.add(_storage.values.toList());
+  }
+}
+
+class MockEntryApi implements EntryApi {
+  Map<String, Entry> _storage = {};
+  StreamController<List<Entry>> _streamController =
+      StreamController<List<Entry>>.broadcast();
+
+  @override
+  Future<Entry> delete(String itemId, String id) async {
+    _broadcast();
```

Comment:
Would this be broadcasting old data, since the remove is a line below?

Reply:
no, adding to a StreamController [fires the events later](https://api.flutter.dev/flutter/dart-async/StreamController/StreamController.broadcast.html): 

> If sync is false, the event will always be fired at a later time, after the code adding the event has completed.

Reply:
Right, but isn't the data going into the call to `add` going to be stale, since the call to `remove` comes after `toList` is making a new copy of the values list?

Reply:
Right, but isn't the data going into the call to `add` going to be stale, since the call to `remove` comes after `toList` is making a new copy of the values list?